Source/UI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -177,13 +177,17 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
             editDelegate?.didFinishEditingBillingForm(successfully: false)
             return
         }
+        var areAllFieldsFulfilled = true
 
-        let areAllFieldFulfilled = style.cells.first(where: {
-          guard let style = $0.style, style.isMandatory else { return false }
-          return textValueOfCellType[$0.index]?.isEmpty == true
```

Comment:
All the fix that was needed was to swap the boolean.

```swift
 return textValueOfCellType[$0.index]?.isEmpty == false
```

I don't think the first/for loop is the issue here. We are looking at a line of 
`textValueOfCellType[$0.index]?.isEmpty`, is there any way we could improve that for readability instead?

Reply:
There are not wrong or right problem. Both solutions are right. i will do your solution to unblock PR.

Reply:
I have tried your solution, it's not working. it disable done button on remove. please advice

Reply:
Have modified. @ehab-al-cko this is very simple Swift syntax, it would be nice if you actually took a minute to understand it before dismissing it. Again, the many levels of optionality present in this solution and the generic naming that doesn't carry meaning at use site make it hard to understand. Its not the for vs first looping...

I have now modified my suggestion to match the [tested solution on the PR you commented on](https://github.com/checkout/frames-ios/pull/268/files#r961532545)

Reply:
I have tried but it's not fully readable for me when != false and when == false. I would like to keep my solution as it's readable for me.